### PR TITLE
Poc 36

### PR DESCRIPTION
**Pull request by hiqedme**

### CODE REVIEW

File: packages/eid/__tests__/eid.test.js [hunk 1]
```diff
@@ -0,0 +1,7 @@
+'use strict';
+
+const eid = require('..');
+
```
Comment (drizzentic) [inline packages/eid/__tests__/eid.test.js:4]: Rebase with main to get latest changes

File: packages/eid/app/api/routes.ts [hunk 2]
```diff
@@ -1,14 +1,84 @@
...
+      return "success";
+    },
+  },
+
+  {
+    method: "GET",
+    path: "/api/push/cd4_count",
     handler: async function (request, h: ResponseToolkit) {
+
+       console.log("sisi ndio tuko");
```
Comment (drizzentic) [inline packages/eid/app/api/routes.ts:43]: Remove debug statements

File: packages/eid/app/api/routes.ts [hunk 3]
```diff
@@ -1,14 +1,84 @@
...
+  },
+  {
+   
+    method: "POST",
+    path: "/uploads",
+    options: payload1,
+    handler: async (request: any, h) => {
+     // console.log("sisi ndio tuko");
+        // const { payload } = request;
+        // return payload;
```
Comment (drizzentic) [inline packages/eid/app/api/routes.ts:61]: Consider removing commented code

File: packages/eid/app/helpers/validators.ts [hunk 4]
```diff
@@ -1,10 +1,24 @@
...
+
+
+
 export default class Validators {
   checkStatusOfViralLoad(viralLoadPayload: string) {
     let status = 0;
     const hasNumbersOnly = /^[0-9]*(?:\.\d{1,2})?$/;
     const hasLessThanSymbol = /</g;
+
+
```
Comment (drizzentic) [inline packages/eid/app/helpers/validators.ts:21]: Format your code to get rid of this spaces

File: packages/eid/app/helpers/validators.ts [hunk 5]
```diff
@@ -32,5 +46,44 @@ export default class Validators {
       whitePaceVar = param.replace(/\s+/g, "");
```
Comment (drizzentic) [inline packages/eid/app/helpers/validators.ts:46]: rename this variable to whiteSpaceVar

File: packages/eid/app/helpers/validators.ts [hunk 6]
```diff
@@ -32,5 +46,44 @@ export default class Validators {
...
+
   }
+
+  validateCsv(file: any) {
+    // console.log(file.mimetype, file.hapi.headers['content-type']);
+    if (file.hapi.headers["content-type"] !== "text/csv") {
+      return { error: "Invalid file type. Only CSV files are allowed" };
+    }
+    if (file.size > 10000) {
+      //10MB
```
Comment (drizzentic) [inline packages/eid/app/helpers/validators.ts:58]: We will need to check the size allowed by nginx for upload

File: packages/eid/app/models/result.ts [hunk 7]
```diff
@@ -21,6 +21,16 @@ declare module EIDPayloads {
     concept: string;
     obsDatetime: any;
     value: number;
+
```
Comment (drizzentic) [inline packages/eid/app/models/result.ts:24]: Format code to get rid of whitespace

File: packages/eid/app/uploads/file.csv [hunk 8]
```diff
@@ -0,0 +1,3 @@
+Lab ID	Batch #	Worksheet #	Plaform	Patient_CCC_No	Provider Identifier	Testing Lab	Partner	County	Sub County	Facility Name	MFL Code	Order Number	AMRS location	Recency Number	Sex	DOB	Age	PMTCT	Sample Type	Collection_Date	Received Status	Delayed Sample Shipment	Current Regimen	ART Initiation Date	Justification	Date Received	Date Entered	Date of Testing	Date of Approval	Date of Dispatch	Viral_Load	Entered By	Received By	Entry
+1197346	208969	26104	C8800	14110-03353		AMPATH Care Lab Eldoret	KCCB-KARP	Migori	Suna East	St Joseph Mission Hospital	14110				Male	21/10/2008	13	None of above	Frozen Plasma	27/06/2022	Accepted		Other	20/05/2009	Routine VL	29/06/2022	##########	01/07/2022	01/07/2022	01/07/2022	"> 10,000,000 cp/ml"	jared ongaga		Health Facility
+1234142	221072	26545	Panther	14701-06836		AMPATH Care Lab Eldoret	AMPATH Plus	West Pokot	West Pokot	Kapenguria District Hospital	14701				Male	06/12/2021	0.667	None of above	Frozen Plasma	01/09/2022	Accepted		ABC+3TC+DTG	01/09/2022	Baseline	02/09/2022	##########	12/09/2022	12/09/2022	12/09/2022	"> 10,000,000 cp/ml"	daniel		Health Facility
```
Comment (drizzentic) [inline packages/eid/app/uploads/file.csv:3]: Please gitignore this file

File: packages/eid/lib/eid.js [hunk 9]
```diff
@@ -0,0 +1,7 @@
+'use strict';
+
+module.exports = eid;
+
+function eid() {
```
Comment (drizzentic) [inline packages/eid/lib/eid.js:5]: remove this file

File: packages/eid/package.json [hunk 10]
```diff
@@ -33,6 +33,10 @@
   "dependencies": {
     "@amrs-integrations/core": "^0.1.0-alpha.18",
     "@hapi/hapi": "20.2.2",
+
+
+    "moment": "^2.29.4",
+
```
Comment (drizzentic) [inline packages/eid/package.json:39]: Format this file

**Review by drizzentic — CHANGES_REQUESTED** [review]

File: packages/eid/.gitignore [hunk 11]
```diff
@@ -107,5 +107,11 @@ build/
...
 /app/poc_sample_payload.ts
 /app/sample_data.ts
+
+/app/uploads
+
+
 /app/uploads
+
+
 token.txt
```
[the remark this answers is not in the document]
Comment (hiqedme) [reply packages/eid/.gitignore]: done

File: .gitignore [hunk 12]
```diff
@@ -106,4 +106,9 @@ build/
...
 
 /app/poc_sample_payload.ts
-/app/sample_data.ts
\ No newline at end of file
+/app/sample_data.ts
+#all csv files
+*.csv
+
+#all DSstore files
+*.DS_Store
```
Comment (drizzentic) [inline .gitignore:114]: Format and add new line after the end of file for all the files

File: packages/eid/app/services/cd4_extractor.ts [hunk 13]
```diff
@@ -0,0 +1,87 @@
...
+      // Iterate through rows and make POST requests
+      for (let row of rows) {
+        // Get patient UUID using identifier
+        let data: any = row;
+        let getPatient = new GetPatient();
+        let patientUUID: any = await getPatient.getPatientUUIDUsingIdentifier(
+          data.ampath_no
+        );
+
+        // console.log("PATIENT UUID" + patientUUID);
```
Comment (drizzentic) [inline packages/eid/app/services/cd4_extractor.ts:41]: Consider removing this console logs

File: .gitignore (hunk 12 above)
Comment (hiqedme) [reply .gitignore]: added

File: packages/eid/app/services/cd4_extractor.ts (hunk 13 above)
Comment (hiqedme) [reply packages/eid/app/services/cd4_extractor.ts]: removed all unnecessary console logs

**Review by drizzentic — COMMENTED** [review]
LGTM! Just remove the ip address on the URL. Leave it empty

**Review by drizzentic — APPROVED** [review]

**Pull request merged**